lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelAlgorithm.cxx

```diff
@@ -724,11 +724,23 @@ Scalar GeneralLinearModelAlgorithm::maximizeReducedLogLikelihood()
   {
     if (normalizedOptimizationBounds_.getDimension() == 0)
     {
-      const Point lowerPoint(optimizationBounds_.getLowerBound());
-      const Point upperPoint(inputTransformation_(optimizationBounds_.getUpperBound()));
-      const Interval::BoolCollection finiteLower(optimizationBounds_.getFiniteLowerBound());
-      const Interval::BoolCollection finiteUpper(optimizationBounds_.getFiniteUpperBound());
-      normalizedOptimizationBounds_ = Interval(lowerPoint, upperPoint, finiteLower, finiteUpper);
+      normalizedOptimizationBounds_ = optimizationBounds_;
+      // Normalize upper bounds.  This can be done only if all scale parameters are activated, and scale
+      // parameters are the first ones.
+      const Description activeParametersDescription(reducedCovarianceModel_.getParameterDescription());
+      UnsignedInteger count = 0;
+      for (UnsignedInteger i = 0; i == count && i < activeParametersDescription.getSize(); ++i)
+        if (activeParametersDescription[i].compare(0, 6, "scale_") == 0)
```

Comment:
@dbarbier Would it be better to check if the active parameter has an index less than reducedCovarianceModel_.getInputDimension() instead of a comparison with a string? All the covariance models should have first a scale parameter of dimension the input dimension, then an amplitude parameter of dimension the output dimension, then the correlation etc. The DiracCovarianceModel does not follow this rule but it is a mistake. Maybe we could leave the code as is and change it once the DiracCovarianceModel would be fixed.

Otherwise, +1 for me.

Reply:
Better how ? That doesnt cost anything and is done only once. Also I thought we discussed ProductCovarianceModel that must have the scale several times or something ?

Reply:
@regislebrun 
> Would it be better to check if the active parameter has an index less than reducedCovarianceModel_.getInputDimension() instead of a comparison with a string?

Handling of parameters must be improved; at the moment Product and Tensorized covariance models do not take into accound specific parameters of other models, if any.  I do not know how this will change, so I chose this solution, which is more reliable IMO.

Reply:
@dbarbier I totally agree with you: the handling of parameters must be improved. The meaning of my remark is the following:
  + either we are able to define a set of common parameters among the different covariance models, with a well-defined meaning and size, such as scale, amplitude, output correlation (for a total of n+p+p(p-1)/2 scalars) and additional specific parameters, and in this case I am in favor of enforcing it throughout the code by using indices
  + or we cannot define such common parameters and we get the semantic of what is a scale base on the content of a string. It is not a matter of performance here, but more on semantic.

IMO these points should be discussed in another PR, the current code is ok for me.